File: seoyeon/week4/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    dic = {'R':0, 'T':0, 'C':0, 'F':0, 'J':0, 'M':0, 'A':0, 'N':0}
    
    for i in range(len(choices)) :
        temp = ''
        if choices[i] < 4 :
            temp = survey[i][0]
            dic[temp] += 4-choices[i]
        else :
            temp = survey[i][1]
            dic[temp] += choices[i]-4
            
    if dic['R'] >= dic['T'] : answer += 'R' 
    else : answer += 'T'
    if dic['C'] >= dic['F'] : answer += 'C'
    else : answer += 'F'
    if dic['J'] >= dic['M'] : answer += 'J'
    else : answer += 'M'
    if dic['A'] >= dic['N'] : answer += 'A'
    else : answer += 'N'
    
    return answer

File: SiHyeon/week4/성격유형검사하기.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    map<char, int> types;
    char s_type[] = { 'R', 'T', 'C', 'F', 'J', 'M', 'A', 'N' };
    for (auto s : s_type) //초기화
        types.insert(pair<char, int>(s, 0));
    for (int i = 0; i < survey.size(); i++)
    {
        if (choices[i] >= 4) //점수 확인해서 4점 이상이면 뒤에 타입에 점수 추가
            types[survey[i].at(1)] += choices[i] % 4;
        else
            types[survey[i].at(0)] += 4 - (choices[i] % 4);
    }
    for (int i = 0; i < 8; i += 2)
    {
        //타입별 점수 비교 후 큰 타입 선택
        answer += types[s_type[i]] >= types[s_type[i + 1]] ? s_type[i] : s_type[i + 1];
    }
    return answer;
}